DruidAvaticaHandlerTest transient failure (times out)

another failure...
```
1:40:14 Running io.druid.sql.avatica.DruidAvaticaHandlerTest
21:40:21 Tests run: 22, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 7.331 sec <<< FAILURE! - in io.druid.sql.avatica.DruidAvaticaHandlerTest
21:40:21 testConcurrentQueries(io.druid.sql.avatica.DruidAvaticaHandlerTest)  Time elapsed: 0.21 sec  <<< ERROR!
21:40:21 java.util.concurrent.ExecutionException: 
21:40:21 AvaticaClientRuntimeException: Remote driver error: ISE: Connection[c767347d-44e8-410a-a14f-e5e9492a5e56] already open.. Error -1 (00000) UNKNOWN
21:40:21 
21:40:21 io.druid.java.util.common.ISE: Connection[c767347d-44e8-410a-a14f-e5e9492a5e56] already open.
21:40:21 	at io.druid.sql.avatica.DruidMeta.openDruidConnection(DruidMeta.java:557)
21:40:21 	at io.druid.sql.avatica.DruidMeta.openConnection(DruidMeta.java:111)
21:40:21 	at org.apache.calcite.avatica.remote.LocalService.apply(LocalService.java:285)
21:40:21 	at org.apache.calcite.avatica.remote.Service$OpenConnectionRequest.accept(Service.java:1771)
21:40:21 	at org.apache.calcite.avatica.remote.Service$OpenConnectionRequest.accept(Service.java:1751)
21:40:21 	at org.apache.calcite.avatica.remote.AbstractHandler.apply(AbstractHandler.java:94)
21:40:21 	at org.apache.calcite.avatica.remote.JsonHandler.apply(JsonHandler.java:52)
21:40:21 	at org.apache.calcite.avatica.server.AvaticaJsonHandler.handle(AvaticaJsonHandler.java:130)
21:40:21 	at io.druid.sql.avatica.DruidAvaticaHandler.handle(DruidAvaticaHandler.java:60)
21:40:21 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
21:40:21 	at org.eclipse.jetty.server.Server.handle(Server.java:534)
21:40:21 	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:320)
21:40:21 	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:251)
21:40:21 	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:283)
21:40:21 	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:108)
21:40:21 	at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93)
21:40:21 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:303)
21:40:21 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
21:40:21 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136)
21:40:21 	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671)
21:40:21 	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589)
21:40:21 	at java.lang.Thread.run(Thread.java:745)
21:40:21 
21:40:21 
21:40:21 	at io.druid.sql.avatica.DruidAvaticaHandlerTest.testConcurrentQueries(DruidAvaticaHandlerTest.java:568)
21:40:21 Caused by: org.apache.calcite.avatica.AvaticaClientRuntimeException: Remote driver error: ISE: Connection[c767347d-44e8-410a-a14f-e5e9492a5e56] already open.
21:40:21 	at io.druid.sql.avatica.DruidAvaticaHandlerTest.lambda$testConcurrentQueries$0(DruidAvaticaHandlerTest.java:555)
21:40:21 
21:40:21 
21:40:21 Results :
21:40:21 
21:40:21 Tests in error: 
21:40:21   DruidAvaticaHandlerTest.testConcurrentQueries:568 » Execution AvaticaClientRun...
```

--- a/sql/src/test/java/io/druid/sql/avatica/DruidAvaticaHandlerTest.java
+++ b/sql/src/test/java/io/druid/sql/avatica/DruidAvaticaHandlerTest.java
@@ -96,7 +96,7 @@ public class DruidAvaticaHandlerTest
     @Override
     public int getMaxConnections()
     {
-      return 2;
+      return 3;
     }
 
     @Override

--- a/sql/src/test/java/io/druid/sql/avatica/DruidAvaticaHandlerTest.java
+++ b/sql/src/test/java/io/druid/sql/avatica/DruidAvaticaHandlerTest.java
@@ -96,6 +96,7 @@ public class DruidAvaticaHandlerTest
     @Override
     public int getMaxConnections()
     {
+      // This must match the number of Connection objects created in setUp()
       return 3;
     }
 
@@ -687,7 +688,7 @@ public class DruidAvaticaHandlerTest
 
     final Connection connection3 = DriverManager.getConnection(url);
     connection3.createStatement().close();
-    
+
     final Connection connection4 = DriverManager.getConnection(url);
     Assert.assertTrue(true);
   }

--- a/sql/src/test/java/io/druid/sql/avatica/DruidAvaticaHandlerTest.java
+++ b/sql/src/test/java/io/druid/sql/avatica/DruidAvaticaHandlerTest.java
@@ -667,9 +667,13 @@ public class DruidAvaticaHandlerTest
     final Connection connection2 = DriverManager.getConnection(url);
     final Statement statement2 = connection2.createStatement();
 
-    expectedException.expect(AvaticaClientRuntimeException.class);
-    expectedException.expectMessage("Too many connections, limit is[2]");
     final Connection connection3 = DriverManager.getConnection(url);
+    final Statement statement3 = connection3.createStatement();
+
+    expectedException.expect(AvaticaClientRuntimeException.class);
+    expectedException.expectMessage("Too many connections, limit is[3]");
+
+    final Connection connection4 = DriverManager.getConnection(url);
   }
 
   @Test
@@ -682,6 +686,9 @@ public class DruidAvaticaHandlerTest
     connection2.createStatement().close();
 
     final Connection connection3 = DriverManager.getConnection(url);
+    connection3.createStatement().close();
+    
+    final Connection connection4 = DriverManager.getConnection(url);
     Assert.assertTrue(true);
   }
 

